cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/relational/StringExpressionFactoryType.java

```diff
@@ -23,5 +23,6 @@ public enum StringExpressionFactoryType implements ExpressionFactoryType<String>
   /**
    * Expression factories of this type will take SQL String expressions
    */
-  SQL;
+  SQL,
+  BIGQUERY_SQL;
```

Comment:
Don't introduce a new type. Use capabilities. Type is a really high level thing that mostly defines core attributes like that it's operating on text SQL strings. Capabilities allow more fine-grained control.

Reply:
That's why there can be only 1 type, but many capabilities. Most plugins that need generic SQL (e.g. SQL92) can simply ask for `SQL` expression factory. The ones that need BigQuery overall or for specific functions may ask for BigQuery capability. Also I'd recommend to ask for BigQuery capability only as needed. We may have Spark factory pretty soon that will be `SQL` but not `BigQuery`

Reply:
So any plugins we write should be able to use just `SQL` for any ANSI SQL expressions and `BigQuery` for specific things